tests/functional/test_show.py

```diff
@@ -159,3 +159,14 @@ def test_pip_show_divider(script):
     result = script.pip('show', 'pip', 'initools')
     lines = result.stdout.splitlines()
     assert "---" in lines
+
+
+@pytest.mark.network
+def test_package_name_is_canonicalized(script):
+    script.pip('install', 'py-salesforce')
```

Comment:
I'd prefer if pip avoided a network call (tests are already flaky enough).
You could install https://github.com/pypa/pip/blob/master/tests/data/packages/pip-test-package-0.1.tar.gz instead
`script.pip('install', 'pip-test-package', '-f', data.packages)` should do it (with `def test_package_name_is_canonicalized(script, data)`)